parquet-format-structures/src/main/java/org/apache/parquet/format/Util.java

```diff
@@ -226,11 +347,24 @@ private static InterningProtocol protocol(TIOStreamTransport t) {
     }
   }
 
-  private static void write(TBase<?, ?> tbase, OutputStream to) throws IOException {
+  private static void write(TBase<?, ?> tbase, OutputStream to, BlockCipher.Encryptor encryptor, byte[] AAD) throws IOException {
+    if (null == encryptor) { 
+      try {
+        tbase.write(protocol(to));
+        return;
+      } catch (TException e) {
+        throw new IOException("can not write " + tbase, e);
+      }
+    }
+    // Serialize and encrypt the structure
+    TMemoryBuffer thriftMemoryBuffer = new TMemoryBuffer(100);
```

Comment:
This one implements `Closeable`, can we wrap a try-with-resources around this? And why 100 buffer size? Maybe move this to a constant, instead of having a magic variable.

Reply:
try-with-resources sounds good.
100 is the initial alloc. Usually enough for page headers etc, but will auto-grow for larger ones.
Will make it a constant.